### ISSUE TITLE
Undefined reference to "main"

### ISSUE THREAD

**Issue by oTMMM** [edenhill/librdkafka#3278]
Read the FAQ first: https://github.com/edenhill/librdkafka/wiki/FAQ

Description
===========
I would like to compile on another directory "producer_cpp" on CentOS 8.3 but got an error.
Compiling with` g++ -o producer_cpp.cpp -lrdkafka++`

The errors :

```
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/crt1.o : In function `_start' :
(.text+0x24) : Undifined reference to "main"
collect2 : error:  ld returned 1 exit status
```

**IMPORTANT**: Always try to reproduce the issue on the latest released version (see https://github.com/edenhill/librdkafka/releases), if it can't be reproduced on the latest version the issue has been fixed.

Checklist
=========

**IMPORTANT**: We will close issues where the checklist has not been completed.

Please provide the following information:

 - [x] librdkafka version (release number or git tag): `<1.6.0>`
 - [x] Apache Kafka version: `<2.13--2.7.0>`
 - [ ] librdkafka client configuration: `<REPLACE with e.g., message.timeout.ms=123, auto.reset.offset=earliest, ..>`
 - [x] Operating system: `<Centos 8.3 (x64)>`
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka
 - [ ] Provide broker log excerpts
 - [x] Critical issue: not critical

**Comment by rolandyoung** [comment]
Your gcc command line looks wrong. The -o option is used to name the executable you want to output (overriding the default of a.out). Try `g++ producer_cpp.cpp -lrdkafka++` or `g++ producer_cpp.cpp -o producer -lrdkafka++`

**Comment by oTMMM** [comment]
First, Thanks to answer me.
Okay, so I tried to compile as you told me and the error is a little bit different :
 
```
$ g++ producer_cpp.cpp -o producer -lrdkafka++
/tmp/cc8UK2hX.o: In function `main':
producer_test.cpp:(.text+0x2ad): undefined reference to `RdKafka::Producer::create(RdKafka::Conf const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)'
producer_test.cpp:(.text+0x4d1): undefined reference to `RdKafka::err2str[abi:cxx11](RdKafka::ErrorCode)'
collect2: error: ld returned 1 exit status
```
I tried to compile on Ubuntu 20.04 as well and got the same error.

**Comment by oTMMM** [comment]
> Your gcc command line looks wrong. The -o option is used to name the executable you want to output (overriding the default of a.out). Try `g++ producer_cpp.cpp -lrdkafka++` or `g++ producer_cpp.cpp -o producer -lrdkafka++`

**Comment by oTMMM** [comment]
> Your gcc command line looks wrong. The -o option is used to name the executable you want to output (overriding the default of a.out). Try `g++ producer_cpp.cpp -lrdkafka++` or `g++ producer_cpp.cpp -o producer -lrdkafka++`

First, Thanks to answer me.
Okay, so I tried to compile as you told me and the error is a little bit different :

```
$ g++ producer_cpp.cpp -o producer -lrdkafka++
/tmp/cc8UK2hX.o: In function `main':
producer_test.cpp:(.text+0x2ad): undefined reference to `RdKafka::Producer::create(RdKafka::Conf const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)'
producer_test.cpp:(.text+0x4d1): undefined reference to `RdKafka::err2str[abi:cxx11](RdKafka::ErrorCode)'
collect2: error: ld returned 1 exit status

```
I tried to compile on Ubuntu 20.04 as well and got the same error.
@oTMMM

**Comment by rolandyoung** [comment]
You need to take a little bit of time to understand the GCC command line.
Please see, for example https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html#Link-Options
If libdrkafka++ is not in your current working directory, you will need to use the -L option to allow the compiler to find it.

TL;DR: Any error you see at build time is almost certainly due to a problem with your gcc command line, not the libraries you are using.
